webapp/tests/Controller/Jury/ContestControllerTest.php

```diff
@@ -0,0 +1,204 @@
+            'contest[processBalloons]'      => $radioValues[1],
+            'contest[public]'               => $radioValues[2],
+            'contest[openToAllTeams]'       => $radioValues[3],
+            'contest[enabled]'              => $radioValues[4],
+        ]);
+        foreach(['jury','admin'] as $role)
+        {
+            $this->testContestsOverview($role, 200, $contest);
+        }
+    }
+
+    public function getTimes () : array
+    {
+        $res = [[
+            "2020-01-01 09:00:00 Europe/Amsterdam",
+            "2021-01-01 09:00:00 Europe/Amsterdam",
+            "2022-01-01 09:00:00 Europe/Amsterdam",
+            "2023-01-01 09:00:00 Europe/Amsterdam",
+            "2024-01-01 09:00:00 Europe/Amsterdam",
+            "2025-01-01 09:00:00 Europe/Amsterdam",
+        ]];
+        // Check times in the past and future
+        foreach([-1,1] as $Yi)
+        {
+            $row = [];
+            foreach(['01','05','09','13','17','21'] as $H)
+            {
+                $row[] = sprintf('%s-06-01 %s:00:00 Europe/Amsterdam',
+                    date('Y') - $Yi, $H);
+            }
+            $res[] = $row;
+        }
+        for($i=-6; $i<2; $i++){
+            $row = [];
+            for($j=0; $j<6; $j++){
+                $row[] = sprintf('%s-06-01 01:00:00 Europe/Amsterdam',
+                        date('Y') + $i + $j );
+            }
+            $res[] = $row;
+        }
+        $row = ['-10:15:30.333333'];
+        $row[] = sprintf('%s-06-01 01:00:00 Europe/Amsterdam',
+            date('Y') );
+        $res[] = array_merge($row, ['+1:00','+13:00:30','+14:00','+80:00']);
+        return $res;
+    }
+
+    /**
+     * Data provider used to test if we can add contests
+     * - the to be created contest
+     * - radio button boolean values (Bit string)
+     */
+    public function provideAddContest () : \Generator
+    {
+        //* - The different times of the contest
+        //TODO: Try things like: "admin'--"
+        $timeStrings = $this->getTimes();
+
+        $contests  = [
+            'NewContest',
+            '⛀⛁⛂⛃⛄⛅⛆⛇⛈⛉⛊⛋⛌⛍⛎⛏',
+            'name with spaces',
+            str_repeat("A", 255)
+        ];
+
+        foreach ($contests as $contest) {
+            yield [$contest,'11111', $timeStrings[0]];
+        }
+        // Try all possible values for the radio buttons
+        // TODO: In the future test if the config option also works
+        for($i=0; $i<2**5; $i++)
+        {
+            $bin = sprintf( "%05d", decbin($i));
+            yield ['RadioTest', $bin, $timeStrings[0]];
+        }
+        // Try all timings
+        foreach($timeStrings as $timeString)
+        {
+            yield ['TimeInterval', '11111', $timeString];
+        }
+    }
+
+    /**
+     * Remove the contest
+     */
+    /* TODO: fix this
+    public function testCheckDeleteContest () : void
+    {
+        $contestShort = 'demo';
+        $contestLong = 'Demo contest';
+        $cid = '2';
```

Comment:
Proposal: use the EntityManagerInterface to get the contest with the given short name?

Reply:
Sounds good, do we already do this somewhere? Otherwise it will take me some extra time

Reply:
Maybe not in the tests. I will check when on a bigger device